avar2.md

```diff
@@ -27,24 +27,28 @@ Axis variation table:
 | `uint16`	| `<reserved>`	| Permanently reserved; set to zero. |
 | `uint16`	| `axisCount`	| The number of variation axes for this font. This must be the same number as axisCount in the 'fvar' table. |
 | `SegmentMaps`	| `axisSegmentMaps[axisCount]`	| The segment maps array — one segment map for each axis, in the order of axes specified in the 'fvar' table. |
-| `Offset32To<DeltaSetIndexMap>` | **`axisIdxMap`** | Offset from beginning of the table, to optional DeltaSetIndexMap storing variation index mapping. |
-| `Offset32To<ItemVariationStore>` | **`varStore`** | Offset from beginning of the table, to optional ItemVariationStore storing variations. |
+| `Offset32` | `axisIdxMapOffset` | Offset from beginning of the table to `axisIdxMap`. |
+| `Offset32` | `varStoreOffset` | Offset from beginning of the table to `varStore`. |
+| `DeltaSetIndexMap` | **`axisIdxMap`** | Indexing of axes used in `varStore`. |
+| `ItemVariationStore` | **`varStore`** | List of `variationRegions` and list of `ItemVariationData` structures including `deltaSets`. |
 
-The table format for `avar` table version 2 is the same as `avar` table version 1 followed by two extra members `axisIdxMap` and `varStore`. It is expected that implementations that handle only version 1 with ignore the entire table by checking the `majorVersion` value.
+The table format for `avar` version 2 is the same as `avar` version 1 followed by offsets to two extra structures, `axisIdxMap` and `varStore`, followed by the data for the `axisIdxMap` and `varStore` structures. The `DeltaSetIndexMap` and `ItemVariationStore` formats are given in [OpenType Font Variations Common Table Formats](https://docs.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats).
 
-Delta values are in the range [-2,2). However, in the `ItemVariationData` they are stored as their true value left-shifted 14 bits (i.e. multiplied by 16,384) in one or two bytes, as `int16` (16-bit signed integer) or `int8` (8 bit signed integer). The LONG_WORDS flag should not be set. For the purposes of calculating interpolated delta values, implementations of `ItemVariationStore` can use existing methods of handling `int16` and `int8` values, only at the end dividing by 16,384 to get the actual delta value.
+The `axisIdxMap` structure maps the axis indices implied in `fvar` to indices used in `varStore`.
+
+Delta values are F2DOT14, therefore in the range [-2,2). However, in `ItemVariationData` structures they are stored as their true value left-shifted 14 bits, i.e. as integers in the range [-32768, 32767].
```

Comment:
I think I see what you mean about not mentioning the range. Larger ranges are actually fine, even those needing int32, as long as they are correctly clamped later.

Reply:
It's not just that. I think what gets used to encode them inside the ItemVariationStore is the internal workings of that structure and not our business to specify. Separation of concerns.

Reply:
I’m addressing this by not mentioning F2DOT14 or even left-shifting at all, thus:

> Delta values are stored as if they were signed integers by multiplying their true value by 16384. Thus 1.0 is stored as 16384; -1.0 is stored as -16384.

Reply:
We need to be clear how a delta value of 0.5 is stored in a structure that only refers to integers.

Reply:
References to F2DOT14 and left-shifting removed, replaced with "Delta values are stored as if they were signed integers by multiplying their true value by 16384."

We need to explain here how to read and write these small fixed point values.